run.ps1

```diff
@@ -0,0 +1,34 @@
+# set the base tools directory
+$toolsLocalPath = Join-Path (Get-Location) "Tools"
+$runExe = Join-Path $toolsLocalPath "Microsoft.DotNet.BuildTools.Run\run.exe"
+
+# if the tool doesn't exist then download and run the boot-strapper script
+if ((Test-Path $runExe) -eq 0)
```

Comment:
You are right about the .toolversion updates.  Detecting to re-download bootstrap.ps1 is tricky though as I don't think we want to always hit the network (i.e. if I have done a sync/build I should be able to then go offline and rebuild without any network traffic).  I can change this to detect bootstrap.ps1 and always execute it as that will address the case of .toolversions changing.